edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/ServerWebSocketTransport.cs

```diff
@@ -20,13 +25,31 @@ class ServerWebSocketTransport : TransportBase
         readonly CancellationTokenSource cancellationTokenSource;
         bool socketAborted;
 
-        public ServerWebSocketTransport(WebSocket webSocket, string localEndpoint, string remoteEndpoint, string correlationId)
+        public ServerWebSocketTransport(WebSocket webSocket,
+                                        string localEndpoint,
+                                        string remoteEndpoint,
+                                        string correlationId,
+                                        Option<X509Certificate2> clientCert,
```

Comment:
Suggest breaking this out into 2 ctors - one for cert auth and one for not.

Reply:
While it makes sense if you see this isolation, as a client of this ctor it does not seem very useful on account of the Option for cert and chain.

Reply:
Not a biggie.. but I think it will make it cleaner.. avoids the ForEach/Expect, etc.

Reply:
Done.